docs/import-from-whitehall.md

```diff
@@ -0,0 +1,136 @@
+rake import:whitehall_migration["nuclear-decommissioning-authority","news_article"]
+```
+
+To import multiple document subtypes (e.g. news stories and press releases) from a single publisher (e.g. NDA):
+
+```
+rake import:whitehall_migration["nuclear-decommissioning-authority","news_article","news_story,press_release"]
+```
+
+There is no limit to the number of subtypes (additional values are added comma separated).
+
+It is also possible to import a single document subtype (e.g. press releases) from a single publisher (e.g. NDA):
+
+```
+rake import:whitehall_migration["nuclear-decommissioning-authority","news_article","press_release"]
+```
+
+After running the rake task, a URL will be printed to the console which allows the status of import to be examined.
+
+## Running a migration locally using govuk-docker
+
+Using [govuk-docker](https://github.com/alphagov/govuk-docker), you can run an import locally without needing to replicate data.  In order to do this, you will need to override the [Plek](https://github.com/alphagov/plek) URL for Whitehall and Publishing API to point to integration.  Additionally, you will require [bearer tokens](#getting-bearer-tokens).
+
+Outside the GDS office, you must be connected to the GDS Developer VPN.  Inside the GDS office, you must be connected to the Brattain network (not GovWifi).  This is due to IP address restrictions on the applications that will be connected to.
+
+### Getting bearer tokens
+
+Tokens can be obtained by creating a new API user in Signon.  These allow access to the relevant remote applications without needing to enter a username and password.
+
+1. Navigate to https://signon.integration.publishing.service.gov.uk/api_users
+1. Click 'Create API user'
+1. Enter a name (e.g. your name followed by 'Whitehall Import to Content Publisher Migration Testing') and an email address
+1. Once the user is created, click 'Add application token'
+1. Select 'Whitehall', then add the 'Export data' permission
+1. Copy the Whitehall token - you will not be able to access this again
+1. Click 'Add application token' again
+1. Select 'Publishing API'
+1. Copy the Publishing API token - you will not be able to access this again
+
+These tokens can then be set as environment variables:
+
+```
+export whitehall_bearer_token=<>
+export publishing_api_bearer_token=<>
+```
+
+Replace `<>` with the bearer tokens you have previously obtained.
+
+### Overriding Plek URLs
+
+By default, Plek will point to your local development environment when running rake tasks in govuk-docker.  Therefore, you will need to override the Plek URLs.
+
+For integration:
+
+```
+export publishing_api_url=https://publishing-api.blue.integration.govuk.digital
+export whitehall_url=https://whitehall-admin.integration.publishing.service.gov.uk
+```
+
+For staging:
+
+```
+export publishing_api_url=https://publishing-api.staging.publishing.service.gov.uk
+export whitehall_url=https://whitehall-admin.staging.publishing.service.gov.uk
+```
```

Comment:
Does the `blue.staging.govuk.digital` work for this? it seems rather weird to be using the carrenza URL here and not sure if it's intended to be supported or not. Life might be simpler if we can use the govuk.digital one in case it's a mistake that this different one is available.

Reply:
Unfortunately not.  This can be seen by visiting https://publishing-api.staging.publishing.service.gov.uk/healthcheck which gives a 200, whereas http://publishing-api.blue.staging.govuk.digital/healthcheck times out.

Reply:
If you use the https equivalent of the latter it works.

Reply:
Ah, I didn't see that.  Have updated.